### PR TITLE
add loading of automatic ORM field labels to i18nTextCollector

### PR DESCRIPTION
**Pull request by xini**

### CODE REVIEW

File: src/i18n/TextCollection/i18nTextCollector.php [hunk 1]
```diff
@@ -854,6 +860,49 @@ public function collectFromEntityProviders($filePath, Module $module = null)
...
+     * @param string $filePath
+     * @return array
+     */
+    public function collectFromORM($filePath)
+    {
+        $entities = [];
+        $classes = ClassInfo::classes_for_file($filePath);
+        foreach ($classes as $class) {
+            // Skip non-implementing classes
+            if (!class_exists($class ?? '')) {
```
Comment (michalkleiner) [inline src/i18n/TextCollection/i18nTextCollector.php:875]: Is the null coalesce needed here?

File: src/Core/ClassInfo.php [hunk 2]
```diff
@@ -301,14 +301,14 @@ public static function classImplements($className, $interfaceName)
...
-        $absFilePath = Director::getAbsFile($filePath);
+        $absFilePath = Director::getAbsFile(realpath($filePath));
         $classManifest = ClassLoader::inst()->getManifest();
         $classes = $classManifest->getClasses();
         $classNames = $classManifest->getClassNames();
 
         $matchedClasses = [];
         foreach ($classes as $lowerClass => $compareFilePath) {
-            if (strcasecmp($absFilePath ?? '', $compareFilePath ?? '') === 0) {
+            if (strcasecmp($absFilePath ?? '', realpath($compareFilePath) ?? '') === 0) {
```
Comment (michalkleiner) [inline src/Core/ClassInfo.php:311]: Is this really needed for the feature? I'd feel better if we didn't have to touch unrelated parts.

Comment (xini) [reply src/Core/ClassInfo.php:311]: I am running this on Windows and the paths don't match if they are not cleaned up. I can submit a separate PR for this if you prefer?

**Comment by michalkleiner** [comment]
Since this is an enhancement I feel like this should target `4` and would be in the next minor version.
Can we update an existing test or add a new one for this new feature?

File: src/i18n/TextCollection/i18nTextCollector.php (hunk 1 above)
Comment (xini) [reply src/i18n/TextCollection/i18nTextCollector.php:875]: I don't know. I have copied the base of the method from https://github.com/silverstripe/silverstripe-framework/blob/4.12/src/i18n/TextCollection/i18nTextCollector.php#L818.

**Comment by xini** [comment]
It's an enhancement for 4, but really it's just restoring the old 3.x functionality. Happy to rebase to 4 if you prefer?

**Comment by GuySartorelli** [comment]
Adding functionality should be done in a minor release, even if that functionality was available in a previous major version of the module. Or in other words: yes please target `4`

File: src/Core/ClassInfo.php (hunk 2 above)
Comment (GuySartorelli) [reply src/Core/ClassInfo.php:311]: This should be a separate pr please.

Comment (xini) [reply src/Core/ClassInfo.php:311]: done, see https://github.com/silverstripe/silverstripe-framework/pull/10687

**Comment by xini** [comment]
done, see https://github.com/silverstripe/silverstripe-framework/pull/10688

**Pull request closed without merging**